dataclasses_json/cfg.py

```diff
@@ -1,10 +1,23 @@
 import functools
-from typing import (Callable, Dict, Optional, Union)
+from typing import (Any, Callable, Dict, NamedTuple, Optional, Union)
 
 from marshmallow.fields import Field as MarshmallowField
 
 from dataclasses_json.undefined import Undefined, UndefinedParameterError
 
+ExcludePredicate = Callable[[str, Any], bool]
```

Comment:
i'd favor making a unary predicate that takes in the field value, rather than the both field name and value. my reasoning is the field name is constant across instances vs. the value which varies

Reply:
By supplying both the key and value, the library could acquire an interesting layer of runtime indirection, allowing for externally-defined behavior without altering the dataclass signature.

That being said, if it's not something desired, I'm totally fine updating this to use a unary predicate.

Reply:
> avoids `Enum` or `NamedTuple` altogether
 But here I'd favor to keep it simple with a plain `Callable[[T], bool]`

Reply:
Not quite metaprogramming, but definitely hosting configuration outside of the dataclass itself so one can consolidate logic around `to_dict()` `to_json()` without having to subclass or add additional layers of handling.

An example - if I was returning the dataclass from a web endpoint, and wanted a user to be able to supply which fields they wanted as parameters, I could do something like:

```python
from flask import request

def is_report_field(k, v):
    return k in request.parameters.split(",")

# ...

@dataclass
class Report(DataClassJsonMixin)
    summary: str = field(metadata=config(exclude=is_report_field))
```
While it isn't a use-case I care about, if we were to go with the value-only predicate initially, and then later decide that the key name would be nice to have, this would be difficult to manage without a breaking API change. 

I think that articulates my thinking. Let me know if this is something we don't wish to support - if not, I'll push up a unary-predicate version.

Reply:
Thanks for the example, I think I understand now. I'm not quite compelled it's enough of a use-case to warrant support in the lib, and there's some overhead to users having to write `lambda _, v: ...` vs. just passing in a predicate